lib/ohai/mixin/rackspace_metadata.rb

```diff
@@ -0,0 +1,51 @@
+#
+# Author:: Alexey Karpik <alexey.karpik@rightscale.com>
+# Author:: Cary Penniman (<cary@rightscale.com>)
+# Author:: Peter Schroeter <peter.schroeter@rightscale.com>
+# Author:: Stas Turlo <stanislav.turlo@rightscale.com>
+# Copyright:: Copyright (c) 2010-2014 RightScale Inc
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "resolv"
+require "json"
+
+module ::Ohai::Mixin::RackspaceMetadata
+  def on_windows?
+    RUBY_PLATFORM =~ /windows|cygwin|mswin|mingw|bccwin|wince|emx/
+  end
+
+  def xenstore_command(command, args)
+    if on_windows?
+      xen_store_client_path = '"c:\Program Files\Citrix\XenTools\xenstore_client.exe"'
+      xen_store_client_alt  = "powershell -NoProfile -NonInteractive -InputFormat None -ExecutionPolicy Bypass \"#{File.join(File.dirname(__FILE__), "xenstore_client.ps1")}\""
```

Comment:
So ultimately why is the script being run? Is it to get some output such as a version? Ultimately looking in the registry may be a quicker and more reliable way to do this. Process creation overhead is not great in Windows unfortunately, and while PowerShell.exe has become faster / slimmer in recent Windows versions, in Win 2008R2 it was still a little sluggish, probably due to its reliance on .net.